### PR TITLE
lib: nrf_fuel_gauge: Update nrf fuel gauge

### PR DESCRIPTION
**Pull request by nordic-auko**
Update nrfxlib nrf fuel gauge to v0.11.1,
and adjust sample and test accordingly.

### CODE REVIEW

File: samples/pmic/native/npm1300_fuel_gauge/README.rst [hunk 1]
```diff
@@ -72,6 +77,7 @@ To connect your DK to the nPM1300 EK, complete the following steps:
    * On the **P17** pin header, connect all LEDs with jumpers.
    * On the **P13** pin header, connect **RSET1** and **VSET1** pins with a jumper.
    * On the **P14** pin header, connect **RSET2** and **VSET2** pins with a jumper.
+   * On the **P18** pin header, disconnect **VDDIO** jumper.
```
Comment (WalkingTalkingPotato) [inline samples/pmic/native/npm1300_fuel_gauge/README.rst:80]: I think it's better to add "Remove all existing connections" at the start of this list, the same way as we have in One Button sample: https://github.com/nrfconnect/sdk-nrf/blob/main/samples/pmic/native/npm1300_one_button/README.rst?plain=1#L38

File: samples/pmic/native/npm1300_fuel_gauge/src/main.c [hunk 2]
```diff
@@ -27,10 +52,32 @@ int main(void)
...
+
+	/* Initialise vbus detection status. */
+	struct sensor_value val;
+	int ret = sensor_attr_get(charger, SENSOR_CHAN_CURRENT, SENSOR_ATTR_UPPER_THRESH, &val);
+
+	if (ret < 0) {
+		return false;
+	}
+
+	vbus_connected = (val.val1 != 0) || (val.val2 != 0);
```
[the remark this answers is not in the document]
Comment (nordic-auko) [reply samples/pmic/native/npm1300_fuel_gauge/src/main.c:75]: Yes, I agree. But the driver doesn't this currently. It reads the vbus status register, but does not expose it (https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/sensor/nordic/npm1300_charger/npm1300_charger.c#L301). I plan to set up an upstream PR to expose it.

File: samples/pmic/native/npm1300_fuel_gauge/README.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (nordic-auko) [reply samples/pmic/native/npm1300_fuel_gauge/README.rst:80]: I'm fine with that

**Review by WalkingTalkingPotato — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -504,8 +504,9 @@ Peripheral samples
 
 PMIC samples
 ------------
+* :ref:`npm1300_fuel_gauge` sample:
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:507]:
```suggestion

* :ref:`npm1300_fuel_gauge` sample:
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -504,8 +504,9 @@ Peripheral samples
...
 
-|no_changes_yet_note|
+  * Updated to accommodate API changes in nRF Fuel Gauge library v0.11.1.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:509]:
```suggestion
  
  * Updated to accommodate API changes in nRF Fuel Gauge library v0.11.1.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Pull request merged**